pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -217,19 +217,24 @@ func checkUserPartOfLibvirtGroup() error {
 	return fmt.Errorf("%s not part of libvirtd group", currentUser.Username)
 }
 
-func fixUserPartOfLibvirtGroup() error {
-	logging.Debug("Adding current user to the libvirt group")
-	currentUser, err := user.Current()
-	if err != nil {
-		logging.Debugf("user.Current() failed: %v", err)
-		return fmt.Errorf("Failed to get current user id")
-	}
-	_, _, err = crcos.RunWithPrivilege("add user to libvirt group", "usermod", "-a", "-G", "libvirt", currentUser.Username)
-	if err != nil {
-		return fmt.Errorf("Failed to add user to libvirt group")
+func fixUserPartOfLibvirtGroup(distro linux.OsType) func() error {
+	return func() error {
+		logging.Debug("Adding current user to the libvirt group")
+		currentUser, err := user.Current()
+		if err != nil {
+			logging.Debugf("user.Current() failed: %v", err)
+			return fmt.Errorf("Failed to get current user id")
+		}
+		_, _, err = crcos.RunWithPrivilege("add user to libvirt group", "usermod", "-a", "-G", "libvirt", currentUser.Username)
```

Comment:
Gerard was suggesting making use of `newgrp` https://linux.die.net/man/1/newgrp

Reply:
I tried `newgrp -` and it changed my current directory to my home. I don't think I like this behaviour.

Reply:
It also broke my shell :(

```
➜  crc git:(shell) xdg-open http://redhat.com // works fine
➜  crc git:(shell) newgrp -
➜  ~ xdg-open http://redhat.com
/usr/bin/xdg-open: line 881: www-browser: command not found
/usr/bin/xdg-open: line 881: links2: command not found
/usr/bin/xdg-open: line 881: elinks: command not found
/usr/bin/xdg-open: line 881: links: command not found
/usr/bin/xdg-open: line 881: lynx: command not found
/usr/bin/xdg-open: line 881: w3m: command not found
xdg-open: no method available for opening 'http://redhat.com'
➜  ~ 
```

Reply:
I don't think we need `newgrp -`, just `newgrp libvirt`, which does not change the environment/the current working dir.